src/main/java/org/codehaus/plexus/util/xml/Xpp3Dom.java

```diff
@@ -209,19 +206,26 @@ public void setAttribute( String name, String value )
 
     public Xpp3Dom getChild( int i )
     {
-        return (Xpp3Dom) childList.get( i );
+        return childList.get( i );
     }
 
     public Xpp3Dom getChild( String name )
     {
-        return (Xpp3Dom) childMap.get( name );
+        for ( int i = childList.size() - 1; i >= 0; i-- )
+        {
+            Xpp3Dom child = childList.get( i );
```

Comment:
Can you explain why you reverse-iterate the array and use indices which are discouraged for lists? Why not [this](https://stackoverflow.com/a/2102552/696632)?

Reply:
The reverse iteration is required but I can switch to a listIterator if you prefer.

Reply:
Yes, please. It makes the code cleaner.